### PR TITLE
Facet

### PR DESCRIPTION
**Pull request by airenzp**
## Description

In this PR the facet plot is implemented and a facet button is added to the chart buttons. To test it from any dataset click on the Facet button and select any two terms. A facet table will be displayed listing the number of samples for the categories associated to the row/column. If the cell has no samples nothing is shown. The number of samples is displayed as a link, when clicked it opens the sampleView with the selected samples. Please check.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by creilly8 — APPROVED** [review]

File: client/dom/select2Terms.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { showTermsTree } from '../mass/groups'
 import { Menu } from '#dom/menu'
 
-export function addDynamicScatterForm(tip, app) {
```
Comment (xzhou82) [inline client/dom/select2Terms.js:4]: please add a comment this is an "intermediate" interface that is used by multiple chart buttons and will route to different apps

File: client/mass/charts.js [hunk 2]
```diff
@@ -199,6 +200,14 @@ function getChartTypeList(self, state) {
 				terms: []
 			}
 		},
+		{
+			label: 'Facet',
```
Comment (xzhou82) [inline client/mass/charts.js:204]: rename to Facet Table

File: client/dom/select2Terms.js (hunk 1 above)
Comment (airenzp) [reply client/dom/select2Terms.js:4]: Done

File: client/mass/charts.js (hunk 2 above)
Comment (airenzp) [reply client/mass/charts.js]: Done

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**